app/controllers/api/plan_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  class PlanController < ApiController
+    # POST api/plan
+    def create
+      planHash = params[:plan]
+      planHash[:timeframes_attributes] = planHash[:timeframes]
+      planHash.delete(:timeframes)
+      planHash[:timeframes_attributes].each.with_index do |timeframe, index|
+        planHash[:timeframes_attributes][index][:segments_attributes] = planHash[:timeframes_attributes][index][:segments]
+        planHash[:timeframes_attributes][index].delete(:segments)
+      end
+      planHash.permit!
+      plan = Plan.create!(planHash)
+      respond_to do |format|
+        format.json {
+          render json: {
+            success: true,
```

Comment:
Is returning `success: true` in JSON redundant to the HTTP status code response?
Does `create!` effectively return an error to the client with a relevant HTTP status code? Here is the pattern I typically use: https://github.com/MAPC/massbuilds/blob/develop/rails/app/controllers/developments_controller.rb#L50-L56